_posts/2021-10-27-#22674.md

```diff
@@ -0,0 +1,101 @@
+  [`TxValidationResult`](https://github.com/bitcoin/bitcoin/blob/88fc7950f8db5f13a6b259819aced2e3db7ff4d8/src/consensus/validation.h#L24) types. Most notably, we distinguish between rule violations and local
+policy-based rejections so that we can inform the P2P layer about peer misbehaviors (see
+[`PeerManagerImpl::MaybePunishNodeForTx`](https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L1393-L1418)).
+In a similar vein, this PR distinguishes between `PCKG_BAD` and `PCKG_POLICY`.
+
+* Miners seek to maximize the total transaction fees while ensuring that their blocks are within
+  consensus-enforced weight and sigops limits. To simplify this [2-dimensional
+knapsack](https://en.wikipedia.org/wiki/Knapsack_problem#Multi-dimensional_knapsack_problem) problem,
+in the mempool, virtual size of a transaction is [calculated](https://github.com/bitcoin/bitcoin/blob/94f83534e4b771944af7d9ed0f40746f392eb75e/src/policy/policy.cpp#L282) as ]()the maximum between its [BIP141 serialized
+size](https://github.com/bitcoin/bips/blob/1f0b563738199ca60d32b4ba779797fc97d040fe/bip-0141.mediawiki#transaction-size-calculations) and
+its "sigop weight".
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+
+2. What does `IsChildWithParents()` check? How does it do this?
+
+3. What criteria must a package meet in order to be considered a child-with-unconfirmed-parents
+package? Is it possible to verify without looking at the current chain? Is it possible to verify
+without looking at our mempool?
+
+4. How does this PR implement checking that a package is child-with-unconfirmed-parents? (Hint: code
+   [here](https://github.com/bitcoin-core-review-club/bitcoin/blob/306a0f6f14972d73281a022d67775a5485d563c7/src/validation.cpp#L1244-L1247)).
+Why do we [add the child's
+inputs](https://github.com/bitcoin-core-review-club/bitcoin/blob/306a0f6f14972d73281a022d67775a5485d563c7/src/validation.cpp#L1234-L1239) to `coins_to_uncache` beforehand?
+
+5. Why do we distinguish between `PCKG_BAD` and `PCKG_POLICY`? Within this PR, do we do anything
+differently based on the result type?
+
+6. In what scenarios could the virtual sizes obtained from `GetVirtualTransactionSize()` [here](https://github.com/bitcoin-core-review-club/bitcoin/commit/78d3665a6d9663cdad188243c6be7e8e9e9ece4e#diff-a58e7bb9d9a8a0287c0b7281d99da4e79b6f8c2a5780c24c6d76c14212c48640L981) and
+the `MempoolAcceptResult` be different? (Hint: is it possible for the tx to be different? Is it
+possible for `PreChecks` to calculate the virtual size differently?)
+
+7. Quiz: given a multi-parent-1-child package of P1, P2, and P3 (where P3 is the child and there
+are no dependencies between the parents), which of the following groups of transactions may be in
+the mempool at the end of `ProcessNewPackage()`?
+
+```
```

Comment:
These lines need to be prefixed with three spaces to be considered part of bullet 7. Without those, it breaks the ordered list and 8. below is rendered as 1. for a new list.